meetings/__init__.py

```diff
@@ -67,18 +75,37 @@ async def log_tag(self, tag, evt: MessageEvent) -> None:
             UPDATE meeting_logs SET tag = $3 WHERE meeting_id = $1 AND timestamp = $2
           """
     await self.database.execute(dbq, meeting, timestamp, tag)
+  
+  async def change_topic(self, topic, evt: MessageEvent) -> None:
+    dbq = """
+            UPDATE meetings SET topic = $3 WHERE meeting_id = $1 AND room_id = $2
+          """
+    await self.database.execute(dbq, self.meeting_id(evt.room_id), evt.room_id, topic)
 
-  # Helper: upload a JSON file
-  async def upload_file(self, evt, items, filename):
-    if len(items) == 0:
-      return
-    
-    logs = tuple()
-    for row in items:
-      log = f"{row[1]} <{row[2]}> #{row[3]}"
-      logs += (log,)
-    
-    data = "\n".join(logs).encode("utf-8")
+    # change the topic of the ^topic message to the topic
+    timestamp = evt.timestamp
+    dbq = """
+            UPDATE meeting_logs SET topic = $3 WHERE meeting_id = $1 AND timestamp = $2
+          """
+    await self.database.execute(dbq, self.meeting_id(evt.room_id), timestamp, topic)
+
+  async def log_to_db(self, meeting, timestamp, sender, message, topic):
+      # Log the item to the db
+      dbq = """
+              INSERT INTO meeting_logs (meeting_id, timestamp, sender, message, topic) VALUES ($1, $2, $3, $4, $5)
+            """
+      await self.database.execute(dbq, meeting, timestamp, sender, message, topic)
+
+  async def send_respond(self, event, message, meeting=None):
+      # could not figure out how to get maubot to work passively on itself, so manually log messages
+      # here instead
+      if meeting and self.config["log_commands"]:
+         await self.log_to_db(self.meeting_id(event.room_id), event.timestamp, self.client.mxid, message, meeting['topic'])
+      await event.respond(message)
+
+  # Helper: upload a file
+  async def upload_file(self, evt, filename, file_contents):
```

Comment:
I'm guessing we should probably move this (and probably a lot of the helpers) to `util.py`, right?